module/VuFindSearch/src/VuFindSearch/Backend/EDS/Base.php

```diff
@@ -447,11 +478,22 @@ protected function call(
             'Content-Type' => $this->contentType,
             'Accept-Encoding' => 'gzip,deflate',
         ];
+        $this->debug(
+            'isguest: ' . $this->isGuest
+            . ' | APIKey: ' . $this->apiKey
+            . ' | APIKey Guest: ' . $this->apiKeyGuest
+        );
         if (null != $headerParams) {
             foreach ($headerParams as $key => $value) {
                 $headers[$key] = $value;
             }
         }
+        if (!empty($this->apiKey)) {
+            $headers['x-api-key'] = $this->apiKey;
+            if ($this->isGuest && !empty($this->apiKeyGuest)) {
```

Comment:
Does this really need to be nested inside the check for the regular API key? Is it possible there could be a scenario where an API key is ONLY used for guests? In that situation, this wouldn't work because the unset non-guest key would prevent the guest key from being examined at all.

Reply:
I think that scenario is actually likely during the initial period where the api key is optional.  Institutions may want to use the guest key for non-authenticated users in order to get that rate limiting behavior (in case of bots) while not affecting authenticated users.  I'm not saying that would work the way the customer expects -- the API may throttle everything anyway -- but I can imagine someone trying it.

Reply:
I'll need to ask Matt :) There are two types of key "legacy" and "new". If a _new_ key has been vended you cannot use EDS API without providing a key. I need to very my assumption that only new keys would be vended going forward and in this case you would need to have an API key and Guest API key would be optional. I'll ask Matt and adjust accordingly

Reply:
If we do end up in the scenario where guest API only works if regular API key is also set, we should add a comment to the .ini file pointing this out to prevent possible confusion in case of misconfiguration.